airbyte-integrations/connectors/source-datascope/source_datascope/source.py

```diff
@@ -0,0 +1,160 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from datetime import datetime, timedelta
+import logging
+import backoff
+
+import requests
+from requests.exceptions import ConnectionError
+from requests.structures import CaseInsensitiveDict
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.sources.streams.http.auth import NoAuth
+
+
+
+# Basic full refresh stream
+class DatascopeStream(HttpStream, ABC):
+
+    # TODO: Fill in the url base. Required.
+    url_base = "https://mydatascope.com/api/external/"
+
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+    @backoff.on_exception(backoff.expo, requests.exceptions.ConnectionError, max_tries=7)
+    def _request(self, url: str, method: str = "GET", data: dict = None) -> List[dict]:
+        response = requests.request(method, url, headers=self._headers, json=data)
+
+        if response.status_code == 200:
+            response_data = response.json()
+            if isinstance(response_data, list):
+                return response_data
+            else:
+                return [response_data]
+        return []
+
+
+class DatascopeIncrementalStream(DatascopeStream, ABC):
+
+    primary_key = "form_answer_id"
+    cursor_field = "updated_at"
+
+    def __init__(self, token: str, form_id: str, start_date: str, schema_type: str, **kwargs):
+        super().__init__()
+        self.token = token
+        self.form_id = form_id
+        self.start_date = start_date
+        self.schema_type = schema_type
+        self.BASE_URL = "https://mydatascope.com"
+        self._headers = {
+            "Authorization": f"Bearer {self.token}",
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        }
+
+    def request_params(
+            self,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        # The api requires that we include the base currency as a query param so we do that in this method
+        return {'token': self.token, 'form_id': self.form_id, 'start': stream_slice['updated_at'], 'date_modified': True, 'order_date': True}
+
+    def get_json_schema(self):
+        schema = super().get_json_schema()
+        if str(self.schema_type) == 'dynamic':
+            return self._request(f"{self.BASE_URL}/api/external/v2/airbyte_schema?token={self.token}&form_id={self.form_id}")[0]
```

Comment:
is `airbyte_schema` a custom endpoint for your instance? how can we generalize this to all users of the connector?

Reply:
The fixed schema will have all the same schema, the dynamic schema will return custom fields depending on the form_id and the questions of that specific form. (configured by the user)

Reply:
I see -- that makes sense. I guess my question is why this specifically calls out `airbyte` in `/v2/airbyte_schema` instead of just `/v2/schema`? maybe just a trivial question, but do you anticipate Airbyte being a core part of Datascope? Sorry if it's a silly question just trying to follow the API convention 😅

Reply:
Hello @sherifnada. I'm with some questions to improve this connector.  It would be awesome if someone from the Airbyte team could help me, I really want to release this connector. Our connector to the DataScope form builder tool right now only work with one form ID, but everything it’s fixed to that ID. 

![Captura de Pantalla 2021-12-13 a la(s) 18 16 24](https://user-images.githubusercontent.com/1148266/145904204-c9de4f65-42f3-419f-ab19-e0a0eb66aa31.png)



 I was wondering how to select multiple forms with only one connection and don't create one connection for each form. My ideal solution would be to select the forms on this step: 
![Captura de Pantalla 2021-12-13 a la(s) 18 04 35](https://user-images.githubusercontent.com/1148266/145903077-c875f38b-f825-4f83-977b-034de7417586.png)

The problem is that right now I have only one fixed model "Form" model and I don't know how to generate multiple schemas on a dynamic way depending the API Key. The endpoints for each model it's the same but the only thing it changes is a param FORM_ID I send to the endpoint. (https://dscope.github.io/docs/#get-all-answers)

Any idea or tip about how to implement this?   Thanks you so much for all the help :)